extra/CMakeLists.txt

```diff
@@ -46,6 +46,19 @@ ADD_CUSTOM_TARGET(GenError
                   DEPENDS
                   ${CMAKE_BINARY_DIR}/include/mysqld_error.h.tmp)
 
+
+ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_BINARY_DIR}/include/mariadbd_options.h
+  COMMAND perl ${PROJECT_SOURCE_DIR}/scripts/generate_option_list.pl
```

Comment:
Perl is not a prerequisite for the build . But CMake is, I'd suggest to take a look at scripting in CMake, it has regular expressions, and not at all weird.

Reply:
CMake's regexes look a lot more basic than what Perl can do, so I don't think that this is feasible

Reply:
well, not everything is solved with regular expression. An alternative is a C++ program, not quite as elegant but does not introduce a required build dependency. On a bright side, CMake does actually know which plugins are compiled, so maybe you would not need that hardcoded list of "plugins that might be there"

Reply:
In C++, we can compile and execute Perl compatible regular expressions fairly easily, with no additional dependency. For an example, you might want to refer to `reg_replace()` in `client/mysqltest.cc`.

Reply:
I just pushed a version written in C++. I don't think that the `--help` parsing can be easily implemented without Perl-style regexes, but they are now used from C++ with PCRE instead.